src/Core/Managed/Shared/Extensibility/MetricAggregatorManager.cs

```diff
@@ -0,0 +1,275 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+#if CORE_PCL || NET45 || NET46
+    using TaskEx = System.Threading.Tasks.Task;
+#endif
+
+    /// <summary>
+    /// Represents a hub for metric aggregation procedures.
+    /// </summary>
+    public sealed class MetricAggregatorManager : IDisposable
+    {
+        /// <summary>
+        /// Name of the property added to aggregation results to indicate duration of the aggregation interval.
+        /// </summary>
+        private static string intervalDurationPropertyName = "IntervalDurationMs";
+
+        /// <summary>
+        /// Reporting frequency.
+        /// </summary>
+        private static TimeSpan snapshotFrequency = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        /// Last time snapshot was initiated.
+        /// </summary>
+        private DateTimeOffset lastSnapshotStartDateTime;
+
+        /// <summary>
+        /// Cancellation token source to allow cancellation of the snapshotting task.
+        /// </summary>
+        private CancellationTokenSource cancellationSource;
+
+        /// <summary>
+        /// Metric aggregator snapshotting task.
+        /// </summary>
+        private Task snapshotTask;
```

Comment:
I think `TaskTimer` fires once, once the `TimeSpan` specified by its `Delay` property elapses. You can adjust the `Delay` property from the `elapsed` function and start again with the new delay. With the benefit of perfect hindsight, this is not the most elegant design, but I think it should get the job done. I do think that timer logic is tricky enough to invest in a single robust implementation that's easy to use for aggregation as well.